### PR TITLE
Feature/bridgeproxy

### PR DESCRIPTION
**Pull request by TheTechArch**
Add proxy for export and trigger batch

### CODE REVIEW

File: src/Altinn.ResourceRegistry.Core/Models/Altinn2/DelegationCountOverview.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿namespace Altinn.ResourceRegistry.Core.Models.Altinn2
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Core/Models/Altinn2/DelegationCountOverview.cs:1]: `#enable nullable` for all new files/classes please

File: src/Altinn.ResourceRegistry.Core/Models/Altinn2/ExportDelegationsRequestBE.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿using System.ComponentModel.DataAnnotations;
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Core/Models/Altinn2/ExportDelegationsRequestBE.cs:1]: `#enable nullable` for all new files/classes please

File: src/Altinn.ResourceRegistry.Core/Models/Altinn2/ExportDelegationsRequestBE.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /// <summary>
+    /// Model to request delegation export
+    /// </summary>
+    public class ExportDelegationsRequestBE
+    {
+        /// <summary>
+        /// Gets or sets the ServiceCode for identifying service
+        /// </summary>
+        [Required]
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Core/Models/Altinn2/ExportDelegationsRequestBE.cs:13]: In addition to `[Required]` attribute, you can do `public required string`. This also solves nullability issues.

File: src/Altinn.ResourceRegistry.Core/Services/ResourceRegistryService.cs [hunk 4]
```diff
@@ -102,7 +102,7 @@ public async Task<List<ServiceResource>> Search(ResourceSearch resourceSearch, C
         /// <inheritdoc/>
         public async Task<List<ServiceResource>> GetSearchResults(ResourceSearch resourceSearch, CancellationToken cancellationToken = default)
         {
-            List<ServiceResource> resourceList = await GetResourceList(false, false, cancellationToken);
+            List<ServiceResource> resourceList = await GetResourceList(false, false, false, cancellationToken);
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Core/Services/ResourceRegistryService.cs:105]: Can you add names to the attributes when you have multiple bools? I have no idea what these `false` values are.

If you change it to something like `GetResourceList(useFoo: false, useBar: false, includeBaz: false, cancellationToken)` it's much easier to understand.

File: src/Altinn.ResourceRegistry.Core/Services/ResourceRegistryService.cs [hunk 5]
```diff
@@ -196,7 +196,7 @@ public async Task<List<ServiceResource>> GetResourceList(bool includeApps, bool
                 
                 if (includeAltinn2)
                 {
-                    tasks.Add(GetAltinn2AvailableServices(orgListTask, cancellationToken));
+                    tasks.Add(GetAltinn2AvailableServices(orgListTask, false, cancellationToken));
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Core/Services/ResourceRegistryService.cs:199]: Same here with regards to the bool names.

File: src/Altinn.ResourceRegistry.Integration/Clients/Altinn2ServicesClient.cs [hunk 6]
```diff
@@ -32,10 +33,10 @@ public Altinn2ServicesClient(HttpClient client, IOptions<PlatformSettings> setti
         }
 
         /// <inheritdoc/>
-        public async Task<List<AvailableService>?> AvailableServices(int languageId, CancellationToken cancellationToken = default)
+        public async Task<List<AvailableService>?> AvailableServices(int languageId, bool includeExpired = default, CancellationToken cancellationToken = default)
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Integration/Clients/Altinn2ServicesClient.cs:36]: Replace `= default` with `= false`.

File: src/Altinn.ResourceRegistry.Integration/Clients/Altinn2ServicesClient.cs [hunk 7]
```diff
@@ -55,6 +56,39 @@ public Altinn2ServicesClient(HttpClient client, IOptions<PlatformSettings> setti
...
+        {
+            string bridgeBaseUrl = _settings.BridgeApiEndpoint;
+            string url = $"{bridgeBaseUrl}authorization/api/delegationexport";
+
+            using HttpContent content = JsonContent.Create(exportDelegationsRequestBE);
+
+            HttpResponseMessage response = await _client.PostAsync(url, content, cancellationToken);
+            response.EnsureSuccessStatusCode();
+
+            return;
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Integration/Clients/Altinn2ServicesClient.cs:70]: Useless return

File: src/Altinn.ResourceRegistry.Integration/Clients/Altinn2ServicesClient.cs [hunk 8]
```diff
@@ -55,6 +56,39 @@ public Altinn2ServicesClient(HttpClient client, IOptions<PlatformSettings> setti
...
+        /// <inheritdoc/>
+        public async Task<DelegationCountOverview?> GetDelegationCount(string serviceCode, int serviceEditionCode, CancellationToken cancellationToken = default)
+        {
+            string bridgeBaseUrl = _settings.BridgeApiEndpoint;
+            string url = $"{bridgeBaseUrl}authorization/api/rights/delegation/delegationcount?serviceCode={serviceCode}&serviceEditionCode={serviceEditionCode}";
+
+            HttpResponseMessage response = await _client.GetAsync(url, cancellationToken);
+            response.EnsureSuccessStatusCode();
+
+            string contentString = await response.Content.ReadAsStringAsync(cancellationToken);
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Integration/Clients/Altinn2ServicesClient.cs:82]: Don't read in response as string. Use the json overloads on the response instead.

File: src/Altinn.ResourceRegistry.Integration/Clients/Altinn2ServicesClient.cs [hunk 9]
```diff
@@ -55,6 +56,39 @@ public Altinn2ServicesClient(HttpClient client, IOptions<PlatformSettings> setti
             }
         }
 
+        /// <inheritdoc/>
+        public async Task ExportDelegations(ExportDelegationsRequestBE exportDelegationsRequestBE, CancellationToken cancellationToken = default)
+        {
+            string bridgeBaseUrl = _settings.BridgeApiEndpoint;
+            string url = $"{bridgeBaseUrl}authorization/api/delegationexport";
+
+            using HttpContent content = JsonContent.Create(exportDelegationsRequestBE);
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry.Integration/Clients/Altinn2ServicesClient.cs:65]: No serializer options required?

File: src/Altinn.ResourceRegistry/Controllers/Altinn2ExportController.cs [hunk 10]
```diff
@@ -65,5 +75,66 @@ await using (var xw = XmlWriter.Create(stream, new XmlWriterSettings { Indent =
 
             return Ok(xsd);
         }
+
+        /// <summary>
+        /// Information about asd
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry/Controllers/Altinn2ExportController.cs:80]: These comments will become part of the swagger we publish. Probably change them to something more reasonable? xD

File: src/Altinn.ResourceRegistry/Controllers/Altinn2ExportController.cs [hunk 11]
```diff
@@ -65,5 +75,66 @@ await using (var xw = XmlWriter.Create(stream, new XmlWriterSettings { Indent =
...
+        /// </summary>
+        [Authorize(Policy = AuthzConstants.POLICY_STUDIO_DESIGNER)]
+        [HttpPost("exportdelegations")]
+        public async Task<ActionResult> ExportDelegations([FromBody] ExportDelegationsRequestBE exportDelegationsRequestBE, CancellationToken cancellationToken = default)
+        {
+            ServiceResource resource = await _resourceRegistryRepository.GetResource(exportDelegationsRequestBE.ResourceId, cancellationToken);
+            if (resource == null)
+            {
+                return BadRequest("Invalid resource");
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry/Controllers/Altinn2ExportController.cs:101]: I suggest using the new `AltinnProblemDetails` made to give out better errors. See the `Altinn.ResourceRegistry.Core.Errors.Problems` class.

File: src/Altinn.ResourceRegistry/Controllers/Altinn2ExportController.cs [hunk 12]
```diff
@@ -65,5 +75,66 @@ await using (var xw = XmlWriter.Create(stream, new XmlWriterSettings { Indent =
...
+            {
+                return BadRequest("No matching resource for org");
+            }
+
+            await _altinn2ServicesClient.ExportDelegations(exportDelegationsRequestBE);
+
+            return Created();
+        }
+
+        private async Task<bool> ValidateMatchingOrgForDelegaton(ExportDelegationsRequestBE exportRequest, string org,  CancellationToken cancellationToken = default)
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry/Controllers/Altinn2ExportController.cs:114]: Add `[NonAction]` attribute to this method. Easier in review when dealing with non-action methods in controllers.

File: src/Altinn.ResourceRegistry/Controllers/Altinn2ExportController.cs [hunk 13]
```diff
@@ -65,5 +75,66 @@ await using (var xw = XmlWriter.Create(stream, new XmlWriterSettings { Indent =
...
+        {
+            List<ServiceResource> altinnService = await _resourceRegistry.GetResourceList(false, true, true, cancellationToken);
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry/Controllers/Altinn2ExportController.cs:116]: Names for the const bools

File: src/Altinn.ResourceRegistry/Controllers/ResourceController.cs [hunk 14]
```diff
@@ -50,7 +51,7 @@ public class ResourceController : ControllerBase
             bool includeAltinn2 = true,
             CancellationToken cancellationToken = default)
         {
-            return await _resourceRegistry.GetResourceList(includeApps, includeAltinn2, cancellationToken);
+            return await _resourceRegistry.GetResourceList(includeApps, includeAltinn2, false,  cancellationToken);
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry/Controllers/ResourceController.cs:54]: Names for the const bools

File: test/Altinn.ResourceRegistry.Tests/Data/Resources/nav_tiltakAvtaleOmArbeidstrening.json [hunk 15]
```diff
@@ -5,11 +5,11 @@
     "en": "Agreement on work training"
   },
   "description": {
-    "nb-NO": "Avtalen skal bidra til at personer med behov for bistand fra NAV skal f� eller beholde en jobb, samtidig som den st�tter arbeidsgivere som inkluderer deg som har nedsatt arbeidsevne. Avtalen inng�s mellom arbeidsgiver, arbeidss�ker og NAV.",
+    "nb-NO": "Avtalen skal bidra til at personer med behov for bistand fra NAV skal f� eller beholde en jobb, samtidig som den st�tter arbeidsgivere som inkluderer deg som har nedsatt arbeidsevne. Avtalen inng�s mellom arbeidsgiver, arbeidss�ker og NAV.",
```
Comment (Alxandr) [inline test/Altinn.ResourceRegistry.Tests/Data/Resources/nav_tiltakAvtaleOmArbeidstrening.json:8]: This seems like a encoding issue

**Review by Alxandr — APPROVED** [review]

File: src/Altinn.ResourceRegistry.Integration/Clients/Altinn2ServicesClient.cs (hunk 9 above)
Comment (TheTechArch) [reply src/Altinn.ResourceRegistry.Integration/Clients/Altinn2ServicesClient.cs]: Added, but seems like default will work based on other clients calling bridge

File: src/Altinn.ResourceRegistry/Program.cs [hunk 16]
```diff
@@ -1,4 +1,6 @@
 using Altinn.ResourceRegistry;
+using Altinn.ResourceRegistry.Core.Constants;
```
Comment (Alxandr) [inline src/Altinn.ResourceRegistry/Program.cs:2]: These are likely not needed.

File: test/Altinn.ResourceRegistry.Tests/Data/Resources/nav_tiltakAvtaleOmArbeidstrening.json [hunk 17]
```diff
@@ -5,11 +5,11 @@
     "en": "Agreement on work training"
   },
   "description": {
-    "nb-NO": "Avtalen skal bidra til at personer med behov for bistand fra NAV skal f� eller beholde en jobb, samtidig som den st�tter arbeidsgivere som inkluderer deg som har nedsatt arbeidsevne. Avtalen inng�s mellom arbeidsgiver, arbeidss�ker og NAV.",
+    "nb-NO": "Avtalen skal bidra til at personer med behov for bistand fra NAV skal få eller beholde en jobb, samtidig som den støtter arbeidsgivere som inkluderer deg som har nedsatt arbeidsevne. Avtalen inngås mellom arbeidsgiver, arbeidssøker og NAV.",
     "EN": "The agreement will help people who need assistance from NAV to get or keep a job, while supporting employers who include those who have reduced working capacity. The agreement is entered into between the employer, the jobseeker and NAV."
   },
   "rightDescription": {
-    "nb-NO": "Med denne fullmakten kan man inng� en avtale om arbeidstreing med NAV og kommune.",
+    "nb-NO": "Med denne fullmakten kan man inng� en avtale om arbeidstreing med NAV og kommune.",
```
Comment (Alxandr) [inline test/Altinn.ResourceRegistry.Tests/Data/Resources/nav_tiltakAvtaleOmArbeidstrening.json:12]: This still seems like an encoding issue

**Pull request merged**